compiler/tests/generated/generic_service_grpc.h

```diff
@@ -35,7 +35,8 @@
 namespace tests
 {
 
-class Foo final
+template <typename Payload>
+    class Foo final
 {
 public:
     template <typename TThreadPool>
```

Comment:
I see that here we anyway deal with a template. Wondering why are the below callbacks use specifically `std::function` and not an arbitrary function object (template)?